internal/provider/virtualmachine/resource_virtualmachine_validator.go

```diff
@@ -84,7 +84,7 @@ func checkKeyPairsInUserData(userdataContent []byte, keyPairs []*harvsterv1.KeyP
 	}
 	var missingKeyPairs []string
 	for _, keyPair := range keyPairs {
-		if ok := inCloudConfig(userData, "ssh_authorized_keys", keyPair.Spec.PublicKey); !ok {
+		if ok := inCloudConfig("", userData, "ssh_authorized_keys", keyPair.Spec.PublicKey); !ok {
```

Comment:
That's a major downside of using `map[interface{}]interface{}`, I like your solution. Maybe in the future we should make use of Go 1.18's new generic types?